### ISSUE TITLE
Typedef warning returned for arrow-parameter in some scenarios even if arrow-parameter is not defined in tslint.json

### ISSUE THREAD

**Issue by maboivin** [palantir/tslint#1273]
### Bug Report

- __TSLint version__: `3.10.2`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: `Visual Studio 2015 / Web Analyzer 1.7.77`

#### TypeScript code being linted

```ts
var o = {
    array: <Array<(p1: string, p2: number) => void>>[]
};

o.array.push((p1, p2) => {
    console.log("The typedef warning occurs on the previous line");
});
```

with `tslint.json`:

```json
{
  "rules": {
    "typedef": [
      true,
      "call-signature",
      "property-declaration",
      "member-variable-declaration",
      "parameter"
    ]
  }
}
```

#### Actual behavior

TSLint returns 2 typedef warnings saying it `expected parameter: 'p1' to have a typedef` and it `expected parameter: 'p2' to have a typedef`. It looks like the arrow-parameter detection is not working in some scenarios.

#### Expected behavior

TSLint doesn't return any warning because the arrow-parameter is not defined for the typedef rule in the tslint.json config file.

**Comment by JKillian** [comment]
@maboivin I can't duplicate this when running the TSLint CLI on the code you provided. My guess is that VS / Web Anaylzer is using an older version of TSLint before the `arrow-parameter` option was introduced. In older versions like that, arrow function parameters were handed by the `parameter` option.

**Comment by maboivin** [comment]
@JKillian You are absolutely right. It looks like with the recent update to Microsoft ASP.NET and Web Tools for VS2015, there are new web linters. There's now an option in Tools / Options / Text Editors / TypeScript / TsLint. I needed to disable TSLint. So the issue is that I had 2 different instances of TSLint running on the same code. As you said, the one included in Microsoft ASP.NET and Web Tools might be an older version because I had already updated the TSLint that ships with Web Analyzer myself by hand to make sure I had the latest version.

This issue is closed.
